### DISCUSSION TITLE
Add custom applications/scripts

### DISCUSSION THREAD

**Discussion by code-on-git** [actions/virtual-environments#9658] [General questions]
Is there any way to add custom scripts or applications when using runner-images?
Ex: I would like to install an internal root cert and register our own powershell repository

**Comment by erik-bershel** [comment]
Hey @code-on-git!
Sure, why not. You may do whatever you wish in your fork.
All you need is to create script to [install](https://github.com/actions/runner-images/blob/main/images/windows/scripts/build/Install-PowershellCore.ps1) or to [configure](https://github.com/actions/runner-images/blob/main/images/windows/scripts/build/Configure-PowerShell.ps1) whatever you wish in an image-branch folder `.\images\<name>\scripts\build` and to add it to template. Somewhere [here](https://github.com/actions/runner-images/blob/main/images/windows/templates/windows-2019.pkr.hcl#L393) in case of windows-2019.

**Reply by code-on-git** [reply]
Thanks,
Off course - that's how I should do it.

Since I couldn't fork it to an Azure Repo I imported a copy into my own repo.
However - I get another issue now, and this has nothing to do with my own custom installation. I don't get that far in the installation. I get stuck on the installation of "Install-VisualStudio.ps1" with the following error:

```
==> azure-arm.image: Provisioning with Powershell...
==> azure-arm.image: Provisioning with powershell script: /azp/_work/5/runner-images/images/windows/templates/../scripts/build/Install-VisualStudio.ps1
    azure-arm.image: Downloading Bootstrapper ...
    azure-arm.image: Downloading package from https://aka.ms/vs/17/release/vs_Enterprise.exe to C:\Users\INSTAL~1\AppData\Local\Temp\vs_Enterprise.exe...
    azure-arm.image: Package downloaded in 0.23 seconds
    azure-arm.image: Signature thumbprint do not match expected.
    azure-arm.image: At C:\Program Files\WindowsPowerShell\Modules\ImageHelpers\InstallHelpers.ps1:972 char:9
    azure-arm.image: +         throw "Signature thumbprint do not match expected."
    azure-arm.image: +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    azure-arm.image:     + CategoryInfo          : OperationStopped: (Signature thumb...match expected.:String) [], RuntimeException
    azure-arm.image:     + FullyQualifiedErrorId : Signature thumbprint do not match expected.
```
